src/black/linegen.py

```diff
@@ -1354,14 +1354,18 @@ def remove_await_parens(node: Node) -> None:
                 if bracket_contents.type != syms.power:
                     ensure_visible(opening_bracket)
                     ensure_visible(closing_bracket)
-                elif (
-                    bracket_contents.type == syms.power
-                    and bracket_contents.children[0].type == token.AWAIT
-                ):
-                    ensure_visible(opening_bracket)
-                    ensure_visible(closing_bracket)
-                    # If we are in a nested await then recurse down.
-                    remove_await_parens(bracket_contents)
+                elif bracket_contents.type == syms.power:
+                    if bracket_contents.children[0].type == token.AWAIT:
+                        ensure_visible(opening_bracket)
+                        ensure_visible(closing_bracket)
+                        # If we are in a nested await then recurse down.
+                        remove_await_parens(bracket_contents)
```

Comment:
When do we need to manually make this recursive call?

Reply:
Seems like we don't, all the tests pass if I remove it. We'll visit nested awaits anyway because this is invoked from a visitor. It also isn't necessary for removing multiple pairs of parens; we have tests for that.